### DISCUSSION TITLE
Change font size via CLI

### DISCUSSION THREAD

**Discussion by mrossinek** [wezterm/wezterm#2550] [Q&A] [answered]
Is it possible, to change the font size of an existing wezterm instance via the CLI?

This would enable integration similar to [this one](https://github.com/folke/zen-mode.nvim/blob/6f5702db4fd4a4c9a212f8de3b7b982f3d93b03c/lua/zen-mode/plugins.lua#L34-L48). The idea there is that when you enter "zen mode" (a focus reading mode plugin for Neovim), the font size of Kitty (in that particular example) gets increased (decreased upon leaving that mode).

**Comment by wez** [comment, marked as answer]
I would suggest using a *user var* named something like ZEN_MODE.

You can set it using something like:

```bash
printf "\033]1337;SetUserVar=%s=%s\007" ZEN_MODE `echo -n on | base64`
``` 

or turn it off:

```bash
printf "\033]1337;SetUserVar=%s=%s\007" ZEN_MODE `echo -n off | base64`
```

In your config, you can handle the zen mode changing using something like this:

```lua
local wezterm = require 'wezterm'

wezterm.on('user-var-changed', function(window, pane, name, value)
  local overrides = window:get_config_overrides() or {}
  if name == "ZEN_MODE" then
     if value == "on" then
       overrides.font_size = 18
     else
       overrides.font_size = nil
    end
  end
  window:set_config_overrides(overrides)  
end)

return {}
```

There are some caveats with this:
* the font size is scoped to the window, but a user-var is scoped to a pane.  There can be multiple tabs and panes associated with a window, so you may wish to add more logic to reconcile what you want to happen when you're working with multiple panes/tabs.
* Cancelling zen mode requires sending a user var in the implementation above, which may result in some frustration if you don't have a convenient way to reset it.  You may wish to add a key binding that clears the overrides.

An alternative implementation might be something like this, which allows you to use the existing/default `CTRL-0` key assignment to reset the font scaling:

```lua
local wezterm = require 'wezterm'

wezterm.on('user-var-changed', function(window, pane, name, value)
  if name == "ZEN_MODE" then
     if value == "on" then
       -- Each IncreaseFontSize will increase by 10%, so if you want it 20% bigger,
       -- call this twice.
       window:perform_action(wezterm.action.IncreaseFontSize, pane)
    end
  end
end)

return {}
```

See: https://wezfurlong.org/wezterm/config/lua/window-events/user-var-changed.html

**Reply by mrossinek** [reply]
Awesome, thanks a lot! Looks like this will be working exactly how I want to (at least in theory). In `vim` this works as expected, but I'm having some trouble figuring out this change of Neovim: https://github.com/neovim/neovim/blob/2afcdbd63a5b0cbeaad9d83b096a3af5201c67a9/runtime/doc/vim_diff.txt#L424-L427
Basically, I need to figure out how I can run the `printf` command from within Neovim such that it affects the shell from which it got started. That `chansend()` suggestion in the link above, does not seem to work for me

**Reply by wez** [reply]
I don't really know nvim, but you may want to borrow this helper function from oscyank:

https://github.com/ojroques/vim-oscyank/blob/f1a0a4b4950842e788722ea542342e2268e0be61/plugin/oscyank.vim#L135-L147

```vim
" Echo a string to the terminal without munging the escape sequences.
function! s:raw_echo(str)
  if has('win32') && has('nvim')
    call chansend(v:stderr, a:str)
  else
    if filewritable('/dev/fd/2')
      call writefile([a:str], '/dev/fd/2', 'b')
    else
      exec("silent! !echo " . shellescape(a:str))
      redraw!
    endif
  endif
endfunction
```

**Reply by mrossinek** [reply]
Thanks, I'll play around with it! :+1:

**Reply by mrossinek** [reply]
Thanks again for your help! In case your curious, here is the result: https://github.com/folke/zen-mode.nvim/pull/61 :+1:
